cluster/manifests/coredns-local/daemonset-coredns.yaml

```diff
@@ -25,6 +25,10 @@ spec:
         component: cluster-dns
       annotations:
         logging/destination: "{{.Cluster.ConfigItems.log_destination_infra}}"
+        prometheus.io/path: /metrics
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9153" # coredns
+        prometheus.io/sidecar-port: "9054" # dnsmasq/unbound
```

Comment:
This is sort of a hack because the pod has two containers exposing metrics. Ideas for how to do this better are welcome!

Reply:
How does this actually work? Wouldn't you get two labels remapped into the same one, so you'd still only get monitoring for one of the containers?

Reply:
what do you think about adding to the config a `source_labels` with `regex: metrics` (or even a more complex regex if needed, like `metrics(-.*)?`) and then define the port metrics on every pod? This way we don't need to specify the port of each container as an annotation.

Reply:
> How does this actually work? Wouldn't you get two labels remapped into the same one, so you'd still only get monitoring for one of the containers?

From my understanding it explodes it into two entities this way. That is what I observed when I tested it.

@jonathanbeber 

Yeah, I also thought about this, maybe it's simpler. I was worried if there would be cases where we need a different port name for some other reason so it couldn't match, but this is only relevant to our pods, so I think it's actually fine.

I would do this if there are no objections?

Reply:
@mikkeloscar +1 from my side

Reply:
Can you define the same port multiple times, with different names? If yes, then it's a very nice solution.

Reply:
@aermakov-zalando I think in this case the little bit more complex regex will be needed.

Reply:
@aermakov-zalando seems like you cannot. When you try to add another port with the same number but different name only the first entry in the list is persisted..

Reply:
Then I'm not sure if I like it, because in a lot of apps metrics would be served from the main port, and I wouldn't want to rename it.

Reply:
Yeah, that would be bad, my suggestion would be renaming it announcing it's also metrics, something like `http-metrics` but it's not clean and weird. From this PR just two applications would actually conflict 

- `kube-static-egress-controller` uses the 8080 but does not define a name for the port, so it wouldn't be renaming;
- `kubernetes-lifecycle-metrics` uses 9090 and also does not define a name, but given the app name I don't think metrics would be a bad name for the port.

Are there other applications that this PR is not considering yet?

Reply:
btw, I'm also fine with using the annotation, I don't think we need to make it more complicated just to avoid one extra annotation here. It's not like it will affect our users, it's just us and our own pods.

Reply:
I don't have a strong opinion. I think it's a bit awkward to name the ports something like metrics, so it's maybe less awkward with the extra annotation for the special case?

Reply:
@mikkeloscar :shipit: let's go for extra annotation